addons/account/wizard/base_document_layout.py

```diff
@@ -1,13 +1,70 @@
-from odoo import models
+from odoo import api, fields, models
 
 
 class BaseDocumentLayout(models.TransientModel):
     _inherit = 'base.document.layout'
 
+    from_invoice = fields.Boolean(compute='_compute_from_invoice', store=False,)
+    qr_code = fields.Boolean(related='company_id.qr_code', readonly=False)
+    vat = fields.Char(related='company_id.vat', readonly=False,)
+    acc_number = fields.Char(compute='_compute_acc_number', inverse='_inverse_acc_number',)
+
     def document_layout_save(self):
         """Save layout and onboarding step progress, return super() result"""
         res = super(BaseDocumentLayout, self).document_layout_save()
         if step := self.env.ref('account.onboarding_onboarding_step_base_document_layout', raise_if_not_found=False):
             for company_id in self.company_id:
                 step.with_company(company_id).action_set_just_done()
+            if res.get('context') is not None:
+                res['context']['dialog_size'] = 'large'
         return res
+
+    @api.depends('report_layout_id')
+    def _compute_from_invoice(self):
+        move = self.env['account.move'].browse(self.env.context.get('active_id'))
+        for record in self:
+            if self.env.context.get('active_model') == 'account.move':
+                record.from_invoice = move.move_type == 'out_invoice'
+            else:
+                record.from_invoice = False
+
+    @api.depends('qr_code')
+    def _compute_preview(self):
+        """
+            This override is needed to add the depends on the qr code and the move
+        """
+        ir_ui_view, wizard_with_logo, preview_css, is_html_empty = super()._compute_preview()
```

Comment:
Following my comment on the parent function, don't do that. Instead, use the wizard record, or import the library. Concerning the `preview_css`, I let you decide if you want to make a helper function or not.